src/oracle/OracleCaller.sol

```diff
@@ -0,0 +1,24 @@
+pragma solidity ^0.8.0;
+
+import { Owned } from "solmate/auth/Owned.sol";
+import { IOracleWriter, Observation } from "src/interfaces/IOracleWriter.sol";
+
+contract OracleCaller is Owned {
+    event WhitelistChanged(address caller, bool whitelist);
+
+    mapping(address => bool) public whitelist;
+
+    constructor(address aOwner) Owned(aOwner) {} // solhint-disable-line no-empty-blocks
+
+    function observation(IOracleWriter aPair, uint256 aIndex) external view returns (Observation memory rObservation) {
+        require(whitelist[msg.sender], "OC: NOT_WHITELISTED");
+        rObservation = aPair.observation(aIndex);
+    }
+
+    function whitelistAddress(address aCaller, bool aWhitelist) external onlyOwner {
+        if (whitelist[aCaller] != aWhitelist) {
```

Comment:
I would remove this check, increases gas cost for smart humans and only save gas for silly humans (with no loss of funds possible)